src/runtime/statistics.py

```diff
@@ -220,6 +220,39 @@ def harmonic_mean(x: list[f64]) -> f64:
 
     return k / sum
 
+
+def mode(x: list[i32]) -> i32:
```

Comment:
We don't have support for `i64` right? Probably generics will be helpful here. Not sure what's the right time to start using them but this is a good use case of using generics along with restrictions.

Reply:
I would start using generics in more tests first, not in the runtime library. I think there will be a lot of things to fix up and improve for generics, especially syntax wise. So we just have to start slowly playing with it and improving it.

Reply:
I see. Let's add a comment here then `TODO: Use generics to support other types`.

Reply:
Done